### PR TITLE
Implement B015

### PR DESCRIPTION
**Pull request by harupy**
#389

### CODE REVIEW

File: src/snapshots/ruff__linter__tests__B015_B015.py.snap [hunk 1]
```diff
@@ -0,0 +1,37 @@
+---
+source: src/linter.rs
+expression: checks
+---
+- kind: UselessComparison
+  location:
+    row: 3
+    column: 2
```
Comment (harupy) [inline src/snapshots/ruff__linter__tests__B015_B015.py.snap:8]: Weird. I think the column should be 0.

Comment (charliermarsh) [reply src/snapshots/ruff__linter__tests__B015_B015.py.snap:8]: I think this is the location of the `==`?

Comment (harupy) [reply src/snapshots/ruff__linter__tests__B015_B015.py.snap:8]: Yes.

File: src/check_ast.rs [hunk 2]
```diff
@@ -1326,6 +1326,12 @@ where
                         self.locate_check(Range::from_located(expr)),
                     ));
                 }
+
+                if self.settings.enabled.contains(&CheckCode::B015) {
+                    if let Some(parent) = self.parents.last() {
+                        flake8_bugbear::plugins::useless_comparison(self, expr, parent);
```
Comment (harupy) [inline src/check_ast.rs:1332]:
```suggestion
                        println!("{:?}", expr);
                        flake8_bugbear::plugins::useless_comparison(self, expr, parent);
```

prints out:

```
Located {
  location: Location { row: 3, column: 2 },
  end_location: Some(Location { row: 3, column: 6 }),
  custom: (),
  node: Compare {
    left: Located { location: Location { row: 3, column: 0 },
    end_location: Some(Location { row: 3, column: 1 }),
    custom: (),
    node: Constant { value: Int(1), kind: None } },
    ops: [Eq], comparators: [Located { location: Location { row: 3, column: 5 }, end_location: Some(Location { row: 3, column: 6 }), custom: (), node: Constant { value: Int(1), kind: None } }] } }
```
```

Comment (charliermarsh) [reply src/check_ast.rs:1332]: Could you use the start location of `node.left` for now?

Comment (harupy) [reply src/check_ast.rs]: Got it. Made the change.

File: src/check_ast.rs [hunk 3]
```diff
@@ -1326,6 +1326,12 @@ where
                         self.locate_check(Range::from_located(expr)),
                     ));
                 }
+
+                if self.settings.enabled.contains(&CheckCode::B015) {
+                    if let Some(parent) = self.parents.last() {
```
Comment (andersk) [inline src/check_ast.rs:1331]: This incorrectly fires for any comparison expression *contained* in an expression statement (e.g. `print(x < y)`), even if it’s not the whole statement.

- #616

Comment (charliermarsh) [reply src/check_ast.rs:1331]: Thanks, good call, that’s on me as reviewer.

**Pull request merged**